src/operations/blas2/gemv.hpp

```diff
@@ -580,7 +581,8 @@ GemvRow<interLoop, Lower, Diag, Upper, Unit, lhs_t, matrix_t, vector_t>::eval(
 
   index_t frs_col = idWFC * dimWFC + interLoop * localid;
   index_t lst_col = std::min(dimC, frs_col + dimWFC);
-  static constexpr value_t init_val = AddOperator::template init<vector_t>();
+  // TODO(Peter): Does it hurt performance if this is not constexpr?
```

Comment:
Can you test that and if it does revert it, If not you can remove the TODO. It is important that we do not introduce performance regression.

Reply:
Wrt. name: originally it was `half` for what the user passes in, and at some point it has to get translated to the C++ type. I've added the translation function here: https://github.com/codeplaysoftware/sycl-blas/pull/251/commits/d6b68d572c46a7302efd412782fa82fd414a80dd

Wrt. performance, we still need to test that.